dbms/src/Operators/ExpressionTransformOp.cpp

```diff
@@ -17,15 +17,31 @@
 
 namespace DB
 {
-OperatorStatus ExpressionTransformOp::transformImpl(Block & block)
+template <bool selective>
+OperatorStatus ExpressionTransformOp<selective>::transformImpl(Block & block)
 {
     if (likely(block))
-        expression->execute(block);
+    {
+        if constexpr (selective)
+        {
+            auto ori_info = block.info;
+            expression->execute(block);
```

Comment:
Don't need to reset `block.info` here?

Reply:
I think it's not necessary, because no ExpressionAction consider `selective_ptr` for now.